firebase_admin/mlkit.py

```diff
@@ -391,6 +444,79 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
+    @classmethod
+    def from_tflite_model_file(cls, model_file_name, bucket_name=None, app=None):
+        """Uploads the model file to an existing Google Cloud Services bucket.
+
+        Args:
+            model_file_name: The name of the model file.
+            bucket_name: The name of an existing bucket. None to use the default bucket configured
+                in the app.
+            app: A Firebase app instance (or None to use the default app).
+
+        Returns:
+            TFLiteGCSModelSource: The source created from the model_file
+
+        Raises:
+            ImportError: If the Cloud Storage Library has not been installed.
+        """
+        gcs_uri = TFLiteGCSModelSource._STORAGE_CLIENT.upload(bucket_name, model_file_name, app)
+        return TFLiteGCSModelSource(gcs_tflite_uri=gcs_uri, app=app)
+
+    @classmethod
+    def from_saved_model(cls, saved_model_dir, bucket_name=None, app=None):
+        """Creates a Tensor Flow Lite model from the saved model, and uploads the model to GCS.
+
+        Args:
+            saved_model_dir: The saved model directory.
+            bucket_name: Optional. The name of the bucket to store the uploaded tflite file.
+                (or None to use the default bucket)
+            app: Optional. A Firebase app instance (or None to use the default app)
+
+        Returns:
+            TFLiteGCSModelSource: The source created from the saved_model_dir
+
+        Raises:
+            ImportError: If the Tensor Flow or Cloud Storage Libraries have not been installed.
+        """
+        if not TF_ENABLED:
+            raise ImportError('Failed to import the tensorflow library for Python. Make sure '
+                              'to install the tensorflow module.')
+        #TODO(ifielker): Do we need to worry about tf version?
+        converter = tf.lite.TFLiteConverter.from_saved_model(saved_model_dir)
```

Comment:
Should we put TF logic inside a class similar to CLoudStorageClient, so these can be unit tested?

Reply:
I thought about it, but ultimately, the functions calling the TF class would be 1-liners, and we would only be testing that we are calling the _TestTFClient instead of the real client which I don't think has sufficient value to bother. I did add a test that shows that it properly throws when tf is not installed. (And I tested the tf functionality manually in Colab).